### PR TITLE
8261937: LambdaForClassInBaseArchive: SimpleApp$$Lambda$1 missing

### PR DESCRIPTION
**Pull request by calvinccheung**
Please review this simple test fix to handle the situation if other lambda proxy classes loaded prior to the one from the test.

The fix passed mach5 tier1 and 2 testing.

Thanks!
Calvin

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8261937](https://bugs.openjdk.java.net/browse/JDK-8261937): LambdaForClassInBaseArchive: SimpleApp$$Lambda$1 missing 

### Reviewers
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**) ⚠️ Review applies to 7682a9a09bf911cb7f80ab97d33406b98aa96781
 * [Daniel D. Daugherty](https://openjdk.java.net/census#dcubed) (@dcubed-ojdk - **Reviewer**) ⚠️ Review applies to 7682a9a09bf911cb7f80ab97d33406b98aa96781

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2897/head:pull/2897`
`$ git checkout pull/2897`

### CODE REVIEW

**Review by iklam — APPROVED** [review]
Looks good. I think this qualifies as a trivial change.

**Review by dcubed-ojdk — APPROVED** [review]

**Comment by calvinccheung** [comment]
@dcubed-ojdk @iklam  Thanks for your review.
I've updated the fix slightly to look for digits after the second `$`.

File: test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/LambdaForClassInBaseArchive.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ private static void doTestCustomBase(String baseArchiveName, String topArchiveNa
               appClass, "lambda")
             .assertNormalExit(out -> {
                     out.shouldHaveExitValue(0)
-                       .shouldContain("Archiving hidden SimpleApp$$Lambda$1");
+                       .shouldMatch("Archiving hidden SimpleApp[$][$]Lambda[$][\\d+]*");
```
Comment (dcubed-ojdk) [inline test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/LambdaForClassInBaseArchive.java:78]: That a zero-or-more digit match or a one-or-more digit match?
I think the pattern should require 1 or more digits.

**Review by dcubed-ojdk — APPROVED** [review]

File: test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/LambdaForClassInBaseArchive.java (hunk 1 above)
Comment (calvinccheung) [reply test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/LambdaForClassInBaseArchive.java:78]: The `\\d+` means one or more digits.

Comment (dcubed-ojdk) [reply test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/LambdaForClassInBaseArchive.java:78]: Thanks for clarifying (for my tired brain).

**Pull request closed without merging**